tests/unit/ip_router_interface/v0/test_ip_router.py

```diff
@@ -0,0 +1,533 @@
+        harness = self._setup()
+
+        # Create a relation
+        rel_id = harness.add_relation("ip-router", "ip-router-requirer")
+        harness.add_relation_unit(rel_id, "ip-router-requirer/0")
+
+        # Update databag
+        network_request = [{"network": "192.168.250.0/24", "gateway": "192.168.250.1"}]
+        harness.update_relation_data(
+            rel_id, "ip-router-requirer", {"networks": json.dumps(network_request)}
+        )
+
+        # Get routing table
+        expected_rt = {"ip-router-requirer": json.dumps(network_request)}
+        assert harness.charm.RouterProvider.get_routing_table() == expected_rt
+        expected_databag = {"networks": json.dumps(network_request)}
+        assert harness.get_relation_data(rel_id, harness.charm.app.name) == expected_databag
+
+    def test_provider_adds_new_network_with_route(self):
+        harness = self._setup()
+
+        # Create a relation
+        rel_id = harness.add_relation("ip-router", "ip-router-requirer")
+        harness.add_relation_unit(rel_id, "ip-router-requirer/0")
+
+        # Update databag with a network with route
+        network_request = [
+            {
+                "network": "192.168.250.0/24",
+                "gateway": "192.168.250.1",
+                "routes": [{"destination": "172.250.0.0/16", "gateway": "192.168.250.3"}],
+            }
+        ]
+        harness.update_relation_data(
+            rel_id, "ip-router-requirer", {"networks": json.dumps(network_request)}
+        )
+
+        # Get routing table
+        expected_rt = {"ip-router-requirer": json.dumps(network_request)}
+        assert harness.charm.RouterProvider.get_routing_table() == expected_rt
+        expected_databag = {"networks": json.dumps(network_request)}
+        assert harness.get_relation_data(rel_id, harness.charm.app.name) == expected_databag
+
+    def test_provider_adds_multiple_networks_from_same_relation(self):
+        harness = self._setup()
+
+        # Create a relation
+        rel_id = harness.add_relation("ip-router", "ip-router-requirer")
+        harness.add_relation_unit(rel_id, "ip-router-requirer/0")
+
+        # Update databag
+        network_request = [
+            {"network": "192.168.250.0/24", "gateway": "192.168.250.1"},
+            {"network": "192.168.251.0/24", "gateway": "192.168.251.1"},
+        ]
+        harness.update_relation_data(
+            rel_id, "ip-router-requirer", {"networks": json.dumps(network_request)}
+        )
+
+        # Get routing table
+        expected_rt = {"ip-router-requirer": json.dumps(network_request)}
+        assert harness.charm.RouterProvider.get_routing_table() == expected_rt
+        expected_databag = {"networks": json.dumps(network_request)}
+        assert harness.get_relation_data(rel_id, harness.charm.app.name) == expected_databag
+
+    def test_provider_adds_multiple_networks_from_multiple_relations(self):
+        harness = self._setup()
```

Comment:
If we use standart setUp method, we do need to create harness in each test method.

Reply:
Done